src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java

```diff
@@ -251,18 +267,94 @@ private String saveRectanglesOnImage(List<Rectangle> rectangles, String filename
                         THICKNESS_LINE);
             }
 
-            filenameResult = filenameImage + "saveRectanglesOnImage.png";
+            filenameResult = pageNumber + "_#4_savedImage.png";
             Imgcodecs.imwrite(filenameResult, matrix);
         }
         return filenameResult;
     }
 
+    private static Rectangle createMainRect(Rectangle rectangle, double offset) {
+        log.debug("offset = " + offset);
+        double width = rectangle.width + offset * 2;
+        double height = rectangle.getHeight();
+        return new Rectangle((int) (rectangle.getX() - offset), rectangle.y, (int) width, (int) height);
+    }
+
     private boolean isTargetRectangle(Rectangle rectangle, int row, int column) {
-        double yLeftUp = rectangle.getY() + OFFSET;
-        double xLeftUp = rectangle.getX() + OFFSET;
-        double xRightDown = xLeftUp + rectangle.getWidth() - OFFSET;
-        double yRightDown = yLeftUp + rectangle.getHeight() - OFFSET;
+        double yLeftUp = rectangle.getY();
+        double xLeftUp = rectangle.getX();
+        double xRightDown = xLeftUp + rectangle.getWidth();
+        double yRightDown = yLeftUp + rectangle.getHeight();
         return xLeftUp < column && column < xRightDown && yLeftUp < row && row < yRightDown;
     }
+
+    private boolean isTargetRect(Rect rectangle, int row, int column) {
+        double yLeftUp = rectangle.y;
+        double xLeftUp = rectangle.x;
+        double xRightDown = xLeftUp + rectangle.width;
+        double yRightDown = yLeftUp + rectangle.height;
+        return xLeftUp <= column && column <= xRightDown && yLeftUp <= row && row <= yRightDown;
+    }
+
+    private double getOffset(Rectangle rectangle, List<double[]> list) {
+        double rightOffset;
+        double leftOffset;
+        double leftRectangleX = rectangle.getX();
+        double rightRectangeX = rectangle.getX() + rectangle.width;
+        double rightListX = list.get(0)[0];
+        double leftListX = list.get(0)[2];
```

Comment:
add X1 and Y2 (or something like this) constants

Reply:
Done.